astarte/device/exceptions.py

```diff
@@ -45,3 +45,69 @@ class APIError(AstarteError):
 
     def __init__(self, body):
         self.body = body
+
+
+class ValidationError(AstarteError):
+    """Exception raised when validation has failed.
+
+    Attributes:
+        msg -- A message error carrying further details
+    """
+
+    def __init__(self, msg):
+        self.msg = msg
+
+
+class PersistencyDirectoryNotFoundError(AstarteError):
+    """Exception raised when the provided persistency directory is not found.
+
+    Attributes:
+        body -- the body of the API reply, which may carry further details
+    """
+
+    def __init__(self, body):
+        self.body = body
+
+
+class InterfaceFileNotFoundError(AstarteError):
+    """Exception raised when a file containing an interface definition does not exists.
+
+    Attributes:
+        msg -- A message error carrying further details
+    """
+
+    def __init__(self, msg):
+        self.msg = msg
+
+
+class InterfaceJsonNotParsableError(AstarteError):
```

Comment:
this could be renamed to 
```suggestion
class InterfaceFileDecodeError(AstarteError):
```
to maintain consistency with the previous one or to
```suggestion
class InterfaceJsonParsingError(AstarteError):
```
to be more readable.